src/main/java/com/commercetools/sync/producttypes/utils/ProductTypeUpdateAttributeDefinitionActionUtils.java

```diff
@@ -161,21 +154,17 @@
                     newAttributesDefinitionsDraftsNameMap.get(oldAttributeDefinitionName);
                 return ofNullable(matchingNewAttributeDefinitionDraft)
                     .map(attributeDefinitionDraft -> {
-                        // attribute type is required so if null we let commercetools to throw exception
-                        if (attributeDefinitionDraft.getAttributeType() != null) {
-                            if (haveSameAttributeType(oldAttributeDefinition, attributeDefinitionDraft)) {
-                                return buildActions(oldAttributeDefinition, attributeDefinitionDraft);
-                            } else {
-                                return Arrays.asList(
-                                    RemoveAttributeDefinition.of(oldAttributeDefinitionName),
-                                    AddAttributeDefinition
-                                        .of(AttributeDefinitionCustomBuilder.of(attributeDefinitionDraft))
-                                );
-                            }
+                        if (haveSameAttributeType(oldAttributeDefinition, attributeDefinitionDraft)) {
+                            return buildActions(oldAttributeDefinition, attributeDefinitionDraft);
                         } else {
-                            return new ArrayList<UpdateAction<ProductType>>();
+                            // this is a work around for changing the type of the definition.
+                            // since there is no action, so we remove then add again.
```

Comment:
@ahmetoz @heshamMassoud What do you think about this change? Is it better or the current version of the comment is better?
```
since there is no way to change an attribute type on CTP, we remove the attribute definition and a new one with a new attribute type
```

Reply:
It looks like the same for me 😄

Reply:
https://github.com/commercetools/commercetools-sync-java/pull/316/commits/93c4d8a4a77f943f8b5edc3a472c34257c57f170

Reply:
Sorry 
`... and add a new ...`